lib/crypto/src/pedersen/mod.rs

```diff
@@ -0,0 +1,115 @@
+//! This module contains Pedersen Hash Function implementation.
+
+pub mod instance;
+pub mod params;
+use alloc::{vec, vec::Vec};
+
+use crate::{
+    arithmetic::{uint::U256, BigInteger},
+    curve::{
+        sw::{Affine, Projective, SWCurveConfig},
+        AffineRepr,
+    },
+    pedersen::params::PedersenParams,
+};
+
+/// Pedersen hash.
+#[derive(Clone, Debug)]
+pub struct Pedersen<F: PedersenParams<P>, P: SWCurveConfig> {
+    params: core::marker::PhantomData<F>,
+    curve: core::marker::PhantomData<P>,
```

Comment:
```suggestion
    params: PhantomData<F>,
    curve: PhantomData<P>,
```
nit: import `PhantomData` directly instead of using absolute path repeatedly